lib/components/user/saved-trip-screen.js

```diff
@@ -0,0 +1,265 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { withLoginRequired } from 'use-auth0-hooks'
+
+import { addTrip, deleteTrip, getTrips, updateTrip } from '../../util/middleware'
+import { routeTo } from '../../actions/ui'
+import DesktopNav from '../app/desktop-nav'
+import { arrayToDayFields, WEEKDAYS } from '../../util/monitored-trip'
+import { getActiveItineraries, getActiveSearch } from '../../util/state'
+import AwaitingScreen from './awaiting-screen'
+import SavedTripEditor from './saved-trip-editor'
+import SavedTripWizard from './saved-trip-wizard'
+import TripBasicsPane from './trip-basics-pane'
+import TripNotificationsPane from './trip-notifications-pane'
+import TripSummaryPane from './trip-summary-pane'
+import withLoggedInUserSupport from './with-logged-in-user-support'
+
+/**
+ * Initializes a monitored trip object from the given query.
+ */
+function getNewMonitoredTrip (loggedInUser, queryParams, itinerary) {
+  return {
+    ...arrayToDayFields(WEEKDAYS),
+    excludeFederalHolidays: true,
+    isActive: true,
+    itinerary,
+    leadTimeInMinutes: 30,
+    queryParams,
+    tripName: '',
+    userId: loggedInUser.id // must provide to API.
```

Comment:
I don't think we should rely on the client to provide the `userId`. Even if they provide this param, the server should overwrite to be sure this doesn't get out of sync/spoofed.

Reply:
Noted in https://github.com/ibi-group/otp-middleware/issues/43 and https://github.com/opentripplanner/otp-react-redux/pull/174/commits/0ce31e2b7ff4192b8d46132b76d6a452414697ee